src/core/fasta_api.h

```diff
@@ -29,10 +29,29 @@
 void gt_fasta_show_entry(const char *description, const char *sequence,
                          GtUword sequence_length, GtUword width, GtFile *outfp);
 
+/* Print a fasta entry with optional <description> and mandatory <sequence> to
+   <outfp>. If <width> is != 0 the sequence is formatted accordingly.
+   Will print at most <sequence_length> characters from <sequence> and at most
+   <description_length> characters from <description> if present. */
```

Comment:
I assume that's because if there is a '\0' in the string before the maximum length is hit, the output will stop there.